tests/unit/test_run.py

```diff
@@ -2371,6 +2387,66 @@ def test_interpolate_single_constant():
         df.interpolate(target)
 
 
+@pytest.mark.parametrize(
+    "years,values",
+    [
+        ([2000, 2001, 2002, 2003], [np.nan, 2.0, 3.0, 4.0]),
+        # We assume Jan 1 values so the following doesn't cover a leap year
+        # [2001, 2002, 2003, 2004],[1.0, 2.0, 3.0, np.nan]),
+        ([2002, 2003, 2004, 2005], [1.0, 2.0, 3.0, np.nan]),
+        ([2003, 2004, 2005, 2006], [1.0, np.nan, 3.0, 4.0]),
+    ],
+)
+def test_interpolate_uniform_years(years, values):
+    df = ScmRun(
+        values,
+        columns={
+            "scenario": ["a_scenario"],
+            "model": ["a_model"],
+            "region": ["World"],
+            "variable": ["Emissions|BC"],
+            "unit": ["Mg /yr"],
+        },
+        index=years,
+    )
+    target_dates = [datetime(y, 1, 1) for y in years]
+
+    res = df.interpolate(
+        target_dates,
+        interpolation_type="linear",
+        extrapolation_type="linear",
+    )
+    with pytest.raises(AssertionError):
+        # difference in last_years before the infilled value causes a small difference
```

Comment:
```suggestion
        # difference in year lengths because of the leap years causes a small difference
```

What is `last_years` ?